Bugfix Metadata & ADT (4)

schmitz-ilias opened the pull request:
This PR fixes [35888](https://mantis.ilias.de/view.php?id=35888), [35515](https://mantis.ilias.de/view.php?id=35515), [35914](https://mantis.ilias.de/view.php?id=35914), [33725](https://mantis.ilias.de/view.php?id=0033725), [35494](https://mantis.ilias.de/view.php?id=35494).

schmitz-ilias committed:
bugfixing in metadata & adt, part 4

schmitz-ilias committed:
fixed copyright notice

alex40724 reviewed (commented):
Thanks for the PR! Looks good to me. @smeyer-ilias If ok for you, you may merge this. One Issue:

Do we need "implements ilAdvancedMetaDataSubItems" for ilObjPortfolio and ilObjPortfolioTemplate as well?

schmitz-ilias committed:
portfolios provide page names to adv md service

schmitz-ilias commented:
@alex40724, good catch, that functionality was indeed missing. I added it to `ilObjPortfolio`, but not to `ilObjPortfolioTemplate`: as  far as I can tell pages in portfolio templates can have assigned custom metadata sets, but those sets can never actually be filled with values (that only happens after a portfolio was created from the template). Since the subitem names are only relevant when already filled values need to be changed, the templates should never have to give out the names of their pages.

Since it's only a small change I added it to this PR, I hope that's alright.

smeyer-ilias reviewed (approved):
Thanks @schmitz-ilias 
btw. I've added the missing type and return types to ilAdvancedMetaDataSubItems
